pkgs/native_assets_cli/lib/src/api/build_output.dart

```diff
@@ -30,6 +34,10 @@ part 'link_output.dart';
 /// hook exists, it will be automatically run, by the Flutter and Dart SDK
 /// tools. The hook is expect to produce a specific output which [BuildOutput]
 /// can produce.
+///
+/// For more information see [build].
+///
+/// Designed to be a sink. [Builder]s stream outputs to the link output.
```

Comment:
> Designed to be a sink

A sink means both the more abstract concept, as here, but also a specific concept in `Stream`s. I would expand that a bit, to not confuse users. ("This means that these are not intended to be read from..." or similar)